controllers/argocd/server/clusterrole.go

```diff
@@ -0,0 +1,127 @@
+package server
+
+import (
+	"reflect"
+
+	"github.com/argoproj-labs/argocd-operator/common"
+	"github.com/argoproj-labs/argocd-operator/pkg/mutation"
+	"github.com/argoproj-labs/argocd-operator/pkg/permissions"
+	v1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (sr *ServerReconciler) reconcileClusterRole() error {
+
+	crName := getClusterRoleName(sr.Instance.Name, sr.Instance.Namespace)
+	crLables := common.DefaultResourceLabels(crName, sr.Instance.Name, ServerControllerComponent)
+
+	// ArgoCD instance is not cluster scoped, cleanup any existing cluster role & exit
+	if !sr.ClusterScoped {
+		return sr.deleteClusterRole(crName)
+	}
+
+	sr.Logger.Info("reconciling clusterRole")
+
+	crRequest := permissions.ClusterRoleRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        crName,
+			Labels:      crLables,
+			Annotations: sr.Instance.Annotations,
+		},
+		Rules:     getPolicyRuleForClusterRule(),
+		Client:    sr.Client,
+		Mutations: []mutation.MutateFunc{mutation.ApplyReconcilerMutation},
+	}
+
+	desiredCR, err := permissions.RequestClusterRole(crRequest)
+	if err != nil {
+		sr.Logger.Error(err, "reconcileClusterRole: failed to request cluster role", "name", desiredCR.Name)
+		sr.Logger.V(1).Info("reconcileClusterRole: one or more mutations could not be applied")
+		return err
```

Comment:
Please update to only returning annotated errors here instead of logging them, as described in the coding practices guide https://github.com/argoproj-labs/argocd-operator/blob/operator-redesign/coding-standards-and-best-practices.md#error-handling
Best to follow that everywhere we are currently logging errors (except at the end of the call stack) 
```suggestion
		sr.Logger.V(1).Info("reconcileClusterRole: one or more mutations could not be applied")
		return errors.Wrapf(err, "reconcileClusterRole: failed to request cluster role %s", desiredCR.Name)
```